.github/workflows/code-checker.yml

```diff
@@ -78,3 +78,15 @@ jobs:
       - name: Run Semgrep Rules
         id: semgrep
         run: semgrep ci --include '*.go' --config 'tools/semgrep/ci'
+
+  gomod:
+    name: Go mod checks
+    runs-on: ubuntu-latest
+    if: github.base_ref == 'main'
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          fetch-depth: 0
```

Comment:
What is the reason for fetching more than one commit?

Reply:
Ahh good question. I copied and pasted the wrong CI check is the answer :-D

Reply:
Great eyes, thanks @Gabrielopesantos!